ncm-2025/completed-notebooks/06_flame_speed_sensitivity_completed.py

```diff
@@ -0,0 +1,170 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.16.1
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# # Flame Speed
+
+# In this example we simulate a freely-propagating, adiabatic, 1-D flame and
+# * Calculate its laminar burning velocity
+# * Perform a sensitivity analysis of its kinetics
+#
+# The figure below illustrates the setup, in a flame-fixed co-ordinate system. The reactants enter with density $\rho_{u}$, temperature $T_{u}$ and speed $S_{u}$. The products exit the flame at speed $S_{b}$, density $\rho_{b}$ and temperature $T_{b}$.
+
+# <img src="../images/flameSpeed.png" alt="Freely Propagating Flame" style="width: 300px;"/>
```

Comment:
Image should be downloaded from the web, and set a white background with border, see examples in other notebooks.